### PR TITLE
Fix create_user implementation

### LINKED ISSUE #171: Mismatch between create_user method implementation and endpoint
### Description

The endpoint for creating a user lists the name field as optional:

https://auth0.com/docs/api/management/v2#!/Users/post_users

But the `create_user` method requires a name parameter:

https://github.com/auth0/ruby-auth0/blob/master/lib/auth0/api/v2/users.rb#L48

I would think the implementation of the gem should match the reality of the endpoint. For instance, perhaps `connection` should be a positional or required named parameter since it is required.

### Prerequisites

* [ ] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
* [ ] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)
* [ ] Did you check the documentation ([repo README](https://github.com/auth0/ruby-auth0/blob/master/README.md) or [Quickstart](https://auth0.com/docs/quickstart/backend/rails))?
* [ ] Did you check the [Auth0 Community](https://community.auth0.com/)?
* [ ] Are you reporting this to the correct repository? See also the [OmniAuth-Auth0 strategy](https://github.com/auth0/omniauth-auth0) for logging in with Rails.
* [ ] Are there any related or duplicate [Issues](https://github.com/auth0/ruby-auth0/issues) or [PRs](https://github.com/auth0/ruby-auth0/pulls) for this issue?

### Environment

* Ruby Auth0 version: master

### PR DESCRIPTION
**Pull request by davidpatrick**
`create_user` currently does not enforce the required connection param, while also forcing an optional param. This change introduces a breaking change since users will need to change their implementation `create_user` 

closes #171

### CODE REVIEW

File: lib/auth0/api/v2/users.rb [hunk 1]
```diff
@@ -43,13 +43,13 @@ def users(options = {})
...
         # @see https://auth0.com/docs/api/v2#!/Users/post_users
-        # @param name [string] The user name.
-        # @param options [hash]
-        #   * :connection [string] The connection the user belongs to.
+        # @param connection [string]  The connection the user belongs to.
+        # @param options [hash] See https://auth0.com/docs/api/management/v2#!/Users/post_users for available options
         # @return [json] Returns the created user.
-        def create_user(name, options = {})
+        def create_user(connection, options = {})
+          raise Auth0::MissingConnection, 'Must supply a valid connection' if connection.to_s.empty?
```
Comment (stevehobbsdev) [inline lib/auth0/api/v2/users.rb:50]: Will this generate a different exception if `connection` is not specified? Just thinking about the `to_s` call on a null object.

Also, should we cover this with tests?

Comment (davidpatrick) [reply lib/auth0/api/v2/users.rb:50]: @stevehobbsdev yes thank you for calling out needed tests, will follow up

File: lib/auth0/api/v2/users.rb [hunk 2]
```diff
@@ -43,13 +43,15 @@ def users(options = {})
...
-        # @param name [string] The user name.
-        # @param options [hash]
-        #   * :connection [string] The connection the user belongs to.
+        # @param connection [string]  The connection the user belongs to.
+        # @param options [hash] See https://auth0.com/docs/api/management/v2#!/Users/post_users for available options
         # @return [json] Returns the created user.
-        def create_user(name, options = {})
+        def create_user(connection, options = {})
+          if !connection.is_a?(String) || connection.is_empty?
+            raise Auth0::MissingParameter, 'Must supply a valid connection'
```
Comment (stevehobbsdev) [inline lib/auth0/api/v2/users.rb:51]: Was it your￼ intention to change the exception type here? It's just in the previous commit it was `MissingConnection`

**Review by stevehobbsdev — APPROVED** [review]
Looks good, just left a small question

File: lib/auth0/api/v2/users.rb (hunk 2 above)
Comment (davidpatrick) [reply lib/auth0/api/v2/users.rb:51]: yes, the missingconnection wasn't common enough to reuse, MissingParameter is sufficient

**Review by stevehobbsdev — APPROVED** [review]

**Pull request merged**